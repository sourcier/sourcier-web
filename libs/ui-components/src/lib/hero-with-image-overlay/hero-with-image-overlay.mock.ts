export const imageUrl = 'https://source.unsplash.com/random';

export const heading = 'Hello there';

export const copy =
  'Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.';
