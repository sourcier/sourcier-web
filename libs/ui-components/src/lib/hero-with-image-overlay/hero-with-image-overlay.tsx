import clsx from 'clsx';

export interface HeroWithImageOverlayProps {
  imageUrl: string;
  heading: string;
  copy: string;
  className?: string;
}

export function HeroWithImageOverlay({
  imageUrl,
  heading,
  copy,
  className,
}: HeroWithImageOverlayProps) {
  return (
    <div
      className={clsx(
        'min-h-[35vh] hero rounded-lg overflow-hidden my-8',
        className
      )}
      style={{
        backgroundImage: `url(${imageUrl})`,
      }}
    >
      <div className="hero-overlay bg-opacity-60"></div>
      <div className="text-center hero-content text-primary-content">
        <div className="max-w-md">
          <h1 className="mb-4 text-5xl font-bold">{heading}</h1>
          {copy}
        </div>
      </div>
    </div>
  );
}

export default HeroWithImageOverlay;
