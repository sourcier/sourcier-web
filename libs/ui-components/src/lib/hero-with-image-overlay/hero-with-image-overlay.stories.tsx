import { Story, Meta } from '@storybook/react';

import {
  HeroWithImageOverlay,
  HeroWithImageOverlayProps,
} from './hero-with-image-overlay';
import { imageUrl, heading, copy } from './hero-with-image-overlay.mock';

export default {
  component: HeroWithImageOverlay,
  title: 'HeroImageOverlay',
} as Meta;

const Template: Story<HeroWithImageOverlayProps> = (args) => (
  <HeroWithImageOverlay {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  imageUrl,
  heading,
  copy,
};
