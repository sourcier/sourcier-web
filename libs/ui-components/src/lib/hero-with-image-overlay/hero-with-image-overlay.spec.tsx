import renderer from 'react-test-renderer';

import HeroWithImageOverlay from './hero-with-image-overlay';
import { imageUrl, heading, copy } from './hero-with-image-overlay.mock';

describe('HeroImageOverlay', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <HeroWithImageOverlay
          copy={copy}
          heading={heading}
          imageUrl={imageUrl}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
