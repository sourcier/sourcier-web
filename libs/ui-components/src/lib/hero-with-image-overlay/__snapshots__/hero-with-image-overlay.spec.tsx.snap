// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImageOverlay should render successfully 1`] = `
<div
  className="min-h-[35vh] hero rounded-lg overflow-hidden my-8"
  style={
    Object {
      "backgroundImage": "url(https://source.unsplash.com/random)",
    }
  }
>
  <div
    className="hero-overlay bg-opacity-60"
  />
  <div
    className="text-center hero-content text-primary-content"
  >
    <div
      className="max-w-md"
    >
      <h1
        className="mb-4 text-5xl font-bold"
      >
        Hello there
      </h1>
      Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
    </div>
  </div>
</div>
`;
